src/app/server/Dnssd.cpp

```diff
@@ -472,44 +473,33 @@ void DnssdServer::StartServer(Dnssd::CommissioningMode mode)
         ChipLogError(Discovery, "Failed to advertise operational node: %s", chip::ErrorStr(err));
     }
 
-    if (HaveOperationalCredentials())
+    if (mode != chip::Dnssd::CommissioningMode::kDisabled)
     {
-        ChipLogProgress(Discovery, "Have operational credentials");
-        if (mode != chip::Dnssd::CommissioningMode::kDisabled)
+        err = AdvertiseCommissionableNode(mode);
+        if (err != CHIP_NO_ERROR)
         {
-            err = AdvertiseCommissionableNode(mode);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(Discovery, "Failed to advertise commissionable node: %s", chip::ErrorStr(err));
-            }
-            // no need to set timeout because callers are currently doing that and their timeout might be longer than the default
+            ChipLogError(Discovery, "Failed to advertise commissionable node: %s", chip::ErrorStr(err));
         }
-#if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
-        else if (GetExtendedDiscoveryTimeoutSecs() != CHIP_DEVICE_CONFIG_DISCOVERY_DISABLED)
+
+        // If any fabrics exist, the commissioning window must have been opened by the administrator
+        // commissioning cluster commands which take care of the timeout.
+        if (!HaveOperationalCredentials())
         {
-            err = AdvertiseCommissionableNode(mode);
-            if (err != CHIP_NO_ERROR)
-            {
-                ChipLogError(Discovery, "Failed to advertise extended commissionable node: %s", chip::ErrorStr(err));
-            }
-            // set timeout
-            ScheduleExtendedDiscoveryExpiration();
+            ScheduleDiscoveryExpiration();
```

Comment:
As a followup, we should re-evalute this `ScheduleDiscoveryExpiration` thing.  If the mode is not kDisabled, then the commissioning window manager opened a commissioning window (whether in response to a cluster command or not, does not matter), and it should be handling its expiration too.  If it is, this code can be removed.  If it's not, it should be fixed and this code can be removed.

Reply:
Yes, by default `OpenCommissioningWindow` is called without timeout, but I agree it would be cleaner if the timeout was set there.

Reply:
> Yes, by default OpenCommissioningWindow is called without timeout

That seems pretty broken too....  Just not advertising via dns-sd but leaving the commissioning window open and still advertising over BLE is .. odd.  Filed https://github.com/project-chip/connectedhomeip/issues/16548

Reply:
All BLEManagers have their own timeout. Actually, the BLE timeouts were added before `CommissioningWindowManager`. I guess that's why it's quite inconsistent.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/16693 on the BLEManager bits too.